### PR TITLE
Refactor and Apply CORS middleware

### PR DESCRIPTION
**Pull request by lumisphere902**
- Small refactor of CORS middleware

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -15,6 +15,16 @@ import (
 // PORT defines where we serve the backend.
 const PORT = ":8081"
 
+func applyMiddlewareCreator(middlewares ...(func(func(w http.ResponseWriter, r *http.Request)) func(w http.ResponseWriter, r *http.Request))) func(func(http.ResponseWriter, *http.Request)) func(http.ResponseWriter, *http.Request) {
```
Comment (lumisphere902) [inline server.go]: I require help on how to make this method signature readable

File: middleware/cors.go [hunk 2]
```diff
@@ -148,18 +146,11 @@ func WithCORSConfig(next http.Handler, c CORSConfig) http.Handler {
...
 
-// WithCORS is middleware that handles your CORS preflight requests quickly
-// and effectively using default settings. It is not verbose. To enable
-// verbosity, please wrap it with some sort of request logging middleware.
-func WithCORS(next http.Handler) http.Handler {
-	// by default we allow all methods, all origins,
-	// and all headers.
-	// MaxAge is omitted, and credentials are not
-	// supported.
+func WithCORS(serveFn func(http.ResponseWriter, *http.Request)) func(http.ResponseWriter, *http.Request) {
```
Comment (krashanoff) [inline middleware/cors.go]: Stripping of descriptive comments on exported functions is bad practice.

File: middleware/cors.go [hunk 3]
```diff
@@ -92,11 +92,9 @@ func (c *CORSConfig) HeadersSupported(requestHeaderFieldNames []string) bool {
...
 }
 
-// WithCORSConfig is middleware that handles your CORS preflight requests quickly
-// and effectively with the supplied configuration. It is not verbose. To enable
-// verbosity, please wrap it with some sort of request logging middleware.
-func WithCORSConfig(next http.Handler, c CORSConfig) http.Handler {
-	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+// Adds CORS functionality to a function, primarily ServeHTTP functions
+func WithCORSConfig(serveFn func(http.ResponseWriter, *http.Request), c CORSConfig) func(http.ResponseWriter, *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
```
Comment (krashanoff) [inline middleware/cors.go]: Exported functions should have comments before their declaration that begin with their name, i.e. `// WithCORSConfig adds CORS functionality...`.

File: server.go (hunk 1 above)
Comment (krashanoff) [reply server.go]: This function, in my opinion, is not needed for our current purposes. While I appreciate the thought of including a function like this for future use, the execution could be smoother and would probably be something we include in our `middleware/` folder. This said, input from the rest of the team on the matter may be desirable.

File: server.go [hunk 4]
```diff
@@ -34,16 +44,19 @@ func main() {
...
+	// set up applyMiddleware
+	applyMiddleware := applyMiddlewareCreator(m.WithCORS)
+
 	// user management
-	router.HandleFunc("/user/get", d.HandleGetUser)
-	router.HandleFunc("/user/update", d.HandleUpdateUser)
-	router.HandleFunc("/user/create", d.HandleInitializeUser)
+	router.HandleFunc("/user/get", applyMiddleware(d.HandleGetUser))
+	router.HandleFunc("/user/update", applyMiddleware(d.HandleUpdateUser))
+	router.HandleFunc("/user/create", applyMiddleware(d.HandleInitializeUser))
```
Comment (krashanoff) [inline server.go]: As mentioned earlier, I recommend that you simply wrap these with `m.WithCORS` for the time being. We do not yet need a wrapper function for large quantities of middleware.

**Pull request merged**